test/trafficsplit/trafficsplit_test.go

```diff
@@ -0,0 +1,263 @@
+package trafficsplit
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+const zeroRPS = "0.0rps"
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+type statTsRow struct {
+	name       string
+	apex       string
+	leaf       string
+	weight     string
+	success    string
+	rps        string
+	latencyP50 string
+	latencyP95 string
+	latencyP99 string
+}
+
+func parseStatTsRow(out string, expectedRowCount, expectedColumnCount int) (map[string]*statTsRow, error) {
+	rows, err := testutil.CheckRowCount(out, expectedRowCount)
+	if err != nil {
+		return nil, err
+	}
+
+	statRows := make(map[string]*statTsRow)
+
+	for _, row := range rows {
+		fields := strings.Fields(row)
+
+		if len(fields) != expectedColumnCount {
+			return nil, fmt.Errorf(
+				"Expected [%d] columns in stat output, got [%d]; full output:\n%s",
+				expectedColumnCount, len(fields), row)
+		}
+
+		row := &statTsRow{
+			name:       fields[0],
+			apex:       fields[1],
+			leaf:       fields[2],
+			weight:     fields[3],
+			success:    fields[4],
+			rps:        fields[5],
+			latencyP50: fields[6],
+			latencyP95: fields[7],
+			latencyP99: fields[8],
+		}
+
+		statRows[row.leaf] = row
+
+	}
+	return statRows, nil
+}
+
+func statTrafficSplit(from string, ns string) (map[string]*statTsRow, error) {
+	time.Sleep(31 * time.Second)
```

Comment:
I'd consider moving this sleep out into the main body of the test, or renaming this function to make it more clear that this will block and wait for old stats to fall out of the window.